@navikt/core/react/src/form/virtualfocus/VirtualFocus.tsx

```diff
@@ -0,0 +1,251 @@
+          const to_focus = to_focus_descendant?.node;
+
+          if (to_focus) {
+            to_focus_descendant.handleOnActive();
+          }
+        } else if (event.key === "ArrowUp") {
+          event.preventDefault();
+          const to_focus_descendant = descendants.prev(virtualFocusIdx, loop);
+          const to_focus = to_focus_descendant?.node;
+
+          if (to_focus) {
+            to_focus_descendant.handleOnActive();
+          }
+        } else if (event.key === "Enter") {
+          const curr = descendants.item(0);
+          if (curr?.handlePick) {
+            curr.handlePick();
+          }
+        }
+      }}
+      {...rest}
+    >
+      {children}
+    </div>
+  );
+});
+
+export const VirtualFocusContent = ({
+  children,
+}: {
+  children: React.ReactNode;
+}) => {
+  const { uniqueId } = useVirtualFocusInternalContext();
+  return (
+    <div
+      id={`virtualfocus-${uniqueId}-content`}
+      className="navds-virtualfocus-content"
+    >
+      {children}
+    </div>
+  );
+};
+
+export interface VirtualFocusItemProps
+  extends React.HTMLAttributes<HTMLDivElement> {
+  /**
+   * The function that is run when the element is focused
+   * (virtually, not actual focus, eg. set a border around an item)
+   */
+  onActive: () => void;
+  /**
+   * The function that is run when the focused element
+   * is to be picked (eg. do an actual search, change route... etc)
+   */
+  pick: () => void;
+  children: React.ReactNode;
+  itemRole?: React.AriaRole;
+}
+
+export const VirtualFocusItem = forwardRef<HTMLElement, VirtualFocusItemProps>(
+  ({ children, onActive, pick, itemRole, ...rest }, ref) => {
+    const { virtualFocusIdx, setVirtualFocusIdx, uniqueId } =
+      useVirtualFocusInternalContext();
+    const { register, descendants, index } = useVirtualFocusDescendant({
+      handleOnActive: () => {
+        setVirtualFocusIdx(index);
+        onActive();
+      },
+      handlePick: () => {
+        pick();
+      },
+    });
+
+    const mergedRefs = useMergeRefs(ref, register);
+
+    return (
+      <div
+        id={`virtualfocus-${uniqueId}-${index}`}
+        className="navds-virtualfocus-item"
+        role={itemRole}
+        data-aksel-virtualfocus={virtualFocusIdx === index}
+        ref={mergedRefs}
+        tabIndex={-1}
+        onKeyDown={() => {}} // Visible, non-interactive elements with click handlers must have at least one keyboard listener
+        onClick={(event) => {
+          const currIdx = descendants.indexOf(event.currentTarget);
+          const curr = descendants.item(currIdx);
+          if (curr) {
+            curr.handlePick();
+          }
+        }}
+        onMouseMove={(event) => {
+          const currIdx = descendants.indexOf(event.currentTarget);
+          const curr = descendants.item(currIdx);
+          if (curr) {
+            setVirtualFocusIdx(curr.index);
+            curr.handleOnActive();
+          }
```

Comment:
Can use `index` from `const { register, descendants, index } = useVirtualFocusDescendant({` here to skip having to look it up in `descendants`
```suggestion
        onMouseMove={(event) => {
            setVirtualFocusIdx(index);
            onActive();
```

Reply:
right, so self reference doesn't need to call the wrapped function inside its descendant item... but I for some reason had this idea of collapsing everything to the same pattern, despite it being "roundabout & inefficient". 🤔 ... but I don't think it saves much on confusion.

Reply:
Yea these changes are purely for performance on my side since it could lead to some unwanted delays if the user swipes the mouse over the list and ends up looping over every item for each mouseOver.